### DISCUSSION TITLE
Unable to build on cloudflare pages

### DISCUSSION THREAD

**Discussion by sazstuff** [chipzoller/hugo-clarity#357] [General]
**I have checked all the prerequisites below and I'm yet experiencing a problem**

- [ x] Read the README.md
- [ x] Have the extended version of Hugo installed
- [ x] Used the exampleSite's config.toml as a reference
- [ x] I have included my repository link. Instead of pasting long snippets here.

**Describe the bug**

Trying to build cloudflare pages with this theme as a sub-module sourced at https://github.com/sazstuff/sazstuff.com and the application build is throwing this error after executing command hugo:

Error: "/opt/buildhome/repo/themes/hugo-clarity/layouts/partials/func/getCodeBlockSettings.html:25:1": parse failed: template: partials/func/getCodeBlockSettings.html:25: function "return" not defined
--

**Desktop (please complete the following information):**
 - OS: [e.g. iOS 14, Windows 10]
 - Browser [e.g. chrome 76, safari 13]

**Comment by sazstuff** [comment]
I ended up switching to just hosting the Hugo built static site instead of trying to have cloudflare build it, but I was hoping to find out why this type of error would show on the cloudflare build but not on my local machine.  Also, is there a better way to ask questions on this theme without posting a bug report?

**Reply by rootwork** [reply]
Hi @sazstuff! I've converted your issue to a discussion post, which is the better place for these kinds of questions.

I'm not very familiar with Cloudflare config, so I'm not sure about the answer to your question, but maybe others will be able to suggest something.

**Reply by sazstuff** [reply]
Thanks!  I am new to github so forgive my ignorance.  I would be interested in this discussion to see what others are using in general to make their sites live.  I tried out cloudflare and it led me down a path of them building the site from source instead of just hosting the site built locally using Hugo.  I ended up making my repository that contains the source code for hugo private and created another repository that is public that has the resultant build from the Public directory.  I just use `hugo -d ..\myotherrepository` and it neatly puts the built files in that folder without the Public subfolder so I can then just do a git commit and push... not too bad, but if there is something even better I am interested!

**Reply by rootwork** [reply]
I'm not sure how it works on Cloudflare, but for instance on GitHub Pages [what they recommend](https://gohugo.io/hosting-and-deployment/hosting-on-github/) is having it in the same repo but then having it "publish" to a dedicated branch. (But of course if you want your source private then the way you have it setup would be the way to go.) 

I assume you've looked at [this guide on Hugo and Cloudflare](https://gohugo.io/hosting-and-deployment/hosting-on-cloudflare-pages/). You might also [ask on the Hugo discussion forums](https://discourse.gohugo.io/). I realize this is a question about the theme, but they might be able to suggest how to get more debugging info from Cloudflare so we could figure out why that theme function isn't being executed properly in that environment.

**Comment by sazstuff** [comment]
Thanks Ivan!  If you don't mind, I have a theme related question and I
honestly have not looked at the discussion forum further to see if there's
an answer to this, but I am having an issue with removing the languages
selector completely (I only speak english).

TLDR; Is there a really quick and easy way to just turn off the additional
languages feature?

I can remove the icon, but you can still mouse over that area and show a
language.  If I remove the reference to the other language, the menu still
pops up with a blank area and you can still select the other language.  If
I remove all of the posts with the language, I get an error.

**Reply by rootwork** [reply]
Did you look at the last section of [this part of the readme](https://github.com/chipzoller/hugo-clarity#i18n)? I think if you follow that then it will remove the icon/mouseover area entirely. But admittedly it's been a long while since I tested that.
